moodlecli/utils.py

```diff
@@ -343,3 +298,78 @@ def _attempts_data_is_stale(attempts, user_id, quiz_id, new_attempt_ts):
             new_grades["attempts"][user_id][quiz_id] = tmp_attempts
 
     return new_grades
+
+
+def course_creation(
+    moodle_client,
+    base_course_id,
+    course_name,
+    course_shortname,
+    course_category,
+    instructor_role_id,
+    instructor_user_id,
+    student_role_id,
+):
+    try:
+        # Create a duplicate course using the base course
+        new_course = moodle_client.copy_course(
+            base_course_id,
+            course_name,
+            course_shortname,
+            course_category,
+        )
+        new_course_id = new_course["id"]
+    except (ConnectionError, Timeout):
+        # Implement work around for Global Accelerator timeout of 340s
+        # (refer to https://github.com/openstax/k12/issues/316 for details)
+        print("Remote disconnected during course copy!")
+        print(f"Polling for course {course_shortname}...")
+        course_found = False
+        while not course_found:
+            sleep(30)
+            print("Querying current courses")
+            data = moodle_client.get_course_by_shortname(
+                course_shortname
+            )
+            if len(data["courses"]) == 0:
+                continue
+            new_course_id = data["courses"][0]["id"]
+            print(f"Found copy: Course ID {new_course_id}")
+            course_found = True
+
+    # Enrol teacher user as a course instructor
+    moodle_client.enrol_user(
+        new_course_id,
+        instructor_user_id,
+        instructor_role_id,
+    )
+
+    # Get enrolment ID for course and student role
+    enrolments = moodle_client.get_self_enrolment_methods(
+        new_course_id, student_role_id
+    )
+    if len(enrolments) != 1:
+        raise Exception(
```

Comment:
You may need to add a `#pragma: no cover` if coverage blocks merging this PR otherwise. We didn't have coverage on this line previously.